src/task_processor.hpp

```diff
@@ -53,9 +53,13 @@ class TaskProcessor {
         task_(task),
         threads_(executor_queue.size()) {
     // Make as many threads as there are executors.
+    threads_rejoined_ = false;
     for (size_t i = 0; i < threads_.size(); i++) {
       threads_[i] = std::thread(&TaskProcessor::thread_handler, this);
     }
+    // Exception pointer is null unless exception occurs on one or more executors/threads.
+    exception_ptr_ = nullptr;
```

Comment:
What do you think about moving these assignments to where these variables were declared?

Reply:
Agreed. Done.